File: java/src/main/java/de/sven_torben/serialization_benchmark/serializer/ISerializer.java
package de.sven_torben.serialization_benchmark.serializer;

import java.io.InputStream;
import java.io.OutputStream;

public interface ISerializer<T> {
	void serialize(OutputStream stream, T data) throws Exception;

	T deserialize(InputStream stream) throws Exception;

	String getName();

	Class<T> getType();
}

File: net/Serializer/ISerializer.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace De.Sven_Torben.Serialization_Benchmark.Serializer
{
    interface ISerializer<T>
    {
        void Serialize(Stream stream, T data);
        T Deserialize(Stream stream);
	    String Name { get; }
        T TestSample { get; }
    }
}
